experimental/table/server/master/src/main/java/alluxio/master/catalog/Table.java

```diff
@@ -14,31 +14,62 @@
 import alluxio.grpc.catalog.ColumnStatisticsInfo;
 import alluxio.grpc.catalog.Schema;
 import alluxio.grpc.catalog.TableInfo;
+import alluxio.grpc.catalog.UdbTableInfo;
+import alluxio.proto.journal.Catalog;
 import alluxio.table.common.udb.UdbTable;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.IOException;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
 
 /**
  * The table implementation which manages all the versions of the table.
  */
 public class Table {
-  private final String mName;
+  private static final Logger LOG = LoggerFactory.getLogger(Table.class);
+
+  private String mName;
   private final Database mDatabase;
-  private final Schema mSchema;
-  private final UdbTable mUdbTable;
+  private Schema mSchema;
+  private UdbTableInfo mTableInfo;
   // TODO(gpang): this should be indexable by partition spec
-  private final ArrayList<Partition> mPartitions;
+  private List<alluxio.grpc.catalog.Partition> mPartitions;
```

Comment:
why is this the grpc partition, and not the java class?

Reply:
when I was constructing a table from the journal, it was not possible for me to construct a HiveLayout, because we don't have dependency on `alluxio.table.under.hive` module.. and therefore I can not construct a java partition class based on that HiveLayout class.